include/jemalloc/internal/extent_inlines.h

```diff
@@ -49,32 +49,20 @@ int extent_snad_comp(const extent_t *a, const extent_t *b);
 #if (defined(JEMALLOC_ENABLE_INLINE) || defined(JEMALLOC_EXTENT_C_))
 JEMALLOC_INLINE arena_t *
 extent_arena_get(const extent_t *extent) {
-	return extent->e_arena;
-}
-
-JEMALLOC_INLINE void *
-extent_base_get(const extent_t *extent) {
-	assert(extent->e_addr == PAGE_ADDR2BASE(extent->e_addr) ||
-	    !extent->e_slab);
-	return PAGE_ADDR2BASE(extent->e_addr);
-}
-
-JEMALLOC_INLINE void *
-extent_addr_get(const extent_t *extent) {
-	assert(extent->e_addr == PAGE_ADDR2BASE(extent->e_addr) ||
-	    !extent->e_slab);
-	return extent->e_addr;
-}
-
-JEMALLOC_INLINE size_t
-extent_size_get(const extent_t *extent) {
-	return extent->e_size;
+	unsigned arena_ind = (unsigned)((extent->e_bits &
+	    EXTENT_BITS_ARENA_MASK) >> EXTENT_BITS_ARENA_SHIFT);
+	if (arena_ind > MALLOCX_ARENA_MAX) {
```

Comment:
I'd imagine reading NULLed code back should be a invalid operation? Maybe remove the brach and assert, and see if it ever happens?